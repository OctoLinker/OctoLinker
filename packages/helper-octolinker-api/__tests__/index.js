import { bulkAction } from '../index';

const response = JSON.stringify({
  result: [
    {
      target: 'bar',
      type: 'npm',
      result: 'https://github.com/foo/bar',
    },
    {
      target: 'i-do-not-exist',
      type: 'npm',
    },
  ],
});

const data = [
  {
    target: 'https://developer.mozilla.org/',
    type: 'ping',
  },
  {
    registry: 'npm',
    target: 'jquery',
    type: 'registry',
  },
  {
    registry: 'npm',
    target: 'i-do-not-exist',
    type: 'registry',
  },
  {
    registry: 'go',
    target: 'jquery',
    type: 'registry',
  },
];

describe('helper-octolinker-api', () => {
  beforeEach(() => {
    fetch.resetMocks();
  });

  it('calls the octolinker api', async () => {
    fetch.mockResponseOnce(response);
    await bulkAction(data);

    expect(global.fetch).toBeCalledWith(expect.any(URL), {
      method: 'GET',
      body: undefined,
      headers: {
        'Content-Type': 'application/json',
      },
    });
  });

  it('sends a GET request', async () => {
    fetch.mockResponseOnce(response);
    await bulkAction(data);

    expect(global.fetch.mock.calls[0][1].method).toEqual('GET');
    expect(global.fetch.mock.calls[0][1].body).toEqual(undefined);
    expect(global.fetch.mock.calls[0][0].href).toEqual(
      'https://octolinker-api.now.sh/?ping=https%3A%2F%2Fdeveloper.mozilla.org%2F&npm=jquery%2Ci-do-not-exist&go=jquery',
    );
  });

  it('sends a POST request if payload is too big for a GET request', async () => {
    const payload = new Array(300)
      .fill({
        registry: 'npm',
        target: 'foo',
        type: 'registry',
      })
      .map((item, index) => ({ ...item, target: item.target + index }));

    fetch.mockResponseOnce(response);
    await bulkAction(payload);

    expect(global.fetch.mock.calls[0][1].method).toEqual('POST');
    expect(global.fetch.mock.calls[0][1].body).toMatchSnapshot();
    expect(global.fetch.mock.calls[0][0].href).toEqual(
      'https://octolinker-api.now.sh/',
    );
  });

  it('returns the result property from the response', async () => {
    fetch.mockResponseOnce(response);

    expect(await bulkAction(data).result).toBe(response.result);
  });

  it('returns an empty array on failure', async () => {
    fetch.mockResponseOnce(() => new Promise((resolve, reject) => reject()));

    expect(await bulkAction(data)).toEqual([]);
  });
});
