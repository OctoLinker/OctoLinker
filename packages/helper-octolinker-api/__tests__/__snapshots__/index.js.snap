// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helper-octolinker-api sends a POST request if payload is too big for a GET request 1`] = `"[{\\"target\\":\\"foo0\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo1\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo2\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo3\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo4\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo5\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo6\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo7\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo8\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo9\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo10\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo11\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo12\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo13\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo14\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo15\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo16\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo17\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo18\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo19\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo20\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo21\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo22\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo23\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo24\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo25\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo26\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo27\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo28\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo29\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo30\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo31\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo32\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo33\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo34\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo35\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo36\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo37\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo38\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo39\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo40\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo41\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo42\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo43\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo44\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo45\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo46\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo47\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo48\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo49\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo50\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo51\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo52\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo53\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo54\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo55\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo56\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo57\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo58\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo59\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo60\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo61\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo62\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo63\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo64\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo65\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo66\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo67\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo68\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo69\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo70\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo71\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo72\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo73\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo74\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo75\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo76\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo77\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo78\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo79\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo80\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo81\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo82\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo83\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo84\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo85\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo86\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo87\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo88\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo89\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo90\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo91\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo92\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo93\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo94\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo95\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo96\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo97\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo98\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo99\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo100\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo101\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo102\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo103\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo104\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo105\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo106\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo107\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo108\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo109\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo110\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo111\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo112\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo113\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo114\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo115\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo116\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo117\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo118\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo119\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo120\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo121\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo122\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo123\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo124\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo125\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo126\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo127\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo128\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo129\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo130\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo131\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo132\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo133\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo134\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo135\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo136\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo137\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo138\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo139\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo140\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo141\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo142\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo143\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo144\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo145\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo146\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo147\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo148\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo149\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo150\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo151\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo152\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo153\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo154\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo155\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo156\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo157\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo158\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo159\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo160\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo161\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo162\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo163\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo164\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo165\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo166\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo167\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo168\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo169\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo170\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo171\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo172\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo173\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo174\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo175\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo176\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo177\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo178\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo179\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo180\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo181\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo182\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo183\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo184\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo185\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo186\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo187\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo188\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo189\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo190\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo191\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo192\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo193\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo194\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo195\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo196\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo197\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo198\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo199\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo200\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo201\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo202\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo203\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo204\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo205\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo206\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo207\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo208\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo209\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo210\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo211\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo212\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo213\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo214\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo215\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo216\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo217\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo218\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo219\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo220\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo221\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo222\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo223\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo224\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo225\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo226\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo227\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo228\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo229\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo230\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo231\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo232\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo233\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo234\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo235\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo236\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo237\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo238\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo239\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo240\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo241\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo242\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo243\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo244\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo245\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo246\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo247\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo248\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo249\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo250\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo251\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo252\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo253\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo254\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo255\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo256\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo257\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo258\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo259\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo260\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo261\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo262\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo263\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo264\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo265\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo266\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo267\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo268\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo269\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo270\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo271\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo272\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo273\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo274\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo275\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo276\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo277\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo278\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo279\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo280\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo281\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo282\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo283\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo284\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo285\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo286\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo287\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo288\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo289\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo290\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo291\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo292\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo293\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo294\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo295\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo296\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo297\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo298\\",\\"type\\":\\"npm\\"},{\\"target\\":\\"foo299\\",\\"type\\":\\"npm\\"}]"`;
